protocol/builder/interface.go

```diff
@@ -0,0 +1,8 @@
+package builder
```

Comment:
Why is this in "builder"? Don't see anything else in here except for the interface either. Can't it just be in the `protocol/api` package?

Reply:
Yep